packages/core/src/schema/models/providerLogs.ts

```diff
@@ -17,41 +18,56 @@ import { latitudeSchema } from '../db-schema'
 import { apiKeys } from '../models/apiKeys'
 import { timestamps } from '../schemaHelpers'
 import { providerApiKeys } from './providerApiKeys'
+import { runErrors } from './runErrors'
 
 export const logSourcesEnum = latitudeSchema.enum('log_source', [
   LogSources.Playground,
   LogSources.API,
   LogSources.Evaluation,
 ])
 
-export const providerLogs = latitudeSchema.table('provider_logs', {
-  id: bigserial('id', { mode: 'number' }).notNull().primaryKey(),
-  uuid: uuid('uuid').notNull().unique(),
-  documentLogUuid: uuid('document_log_uuid'),
-  providerId: bigint('provider_id', { mode: 'number' })
-    .notNull()
-    .references(() => providerApiKeys.id, {
-      onDelete: 'restrict',
-      onUpdate: 'cascade',
-    }),
-  model: varchar('model'),
-  finishReason: varchar('finish_reason').notNull().default('stop'),
-  config: json('config').$type<PartialConfig>().notNull(),
-  messages: json('messages').$type<Message[]>().notNull(),
-  responseObject: jsonb('response_object').$type<unknown>(),
-  responseText: text('response_text').$type<string>(),
-  toolCalls: json('tool_calls').$type<ToolCall[]>().notNull().default([]),
-  tokens: bigint('tokens', { mode: 'number' }).notNull(),
-  costInMillicents: integer('cost_in_millicents').notNull().default(0),
-  duration: bigint('duration', { mode: 'number' }).notNull(), // in milliseconds!
-  source: logSourcesEnum('source').notNull(),
-  apiKeyId: bigint('apiKeyId', { mode: 'number' }).references(
-    () => apiKeys.id,
-    {
-      onDelete: 'restrict',
-      onUpdate: 'cascade',
-    },
-  ),
-  generatedAt: timestamp('generated_at', { mode: 'date' }).notNull(),
-  ...timestamps(),
-})
+export const providerLogs = latitudeSchema.table(
+  'provider_logs',
+  {
+    id: bigserial('id', { mode: 'number' }).notNull().primaryKey(),
+    uuid: uuid('uuid').notNull().unique(),
+    documentLogUuid: uuid('document_log_uuid'),
+    providerId: bigint('provider_id', { mode: 'number' })
+      .notNull()
+      .references(() => providerApiKeys.id, {
+        onDelete: 'restrict',
+        onUpdate: 'cascade',
+      }),
+    errorId: bigint('error_id', { mode: 'number' }).references(
```

Comment:
why does provider log have a pointer to error?

Reply:
An error can happen

1. Before AI is called. 
2. While AI is called. When chunks bring `type: error` or `finish` with `finishReason: 'error'`


This pointer is here to save the error + provider log at the same time with a pointer to the error from the provider log


I can not do this. But the question is should we throw an error when stream finish with `type: error` or `finishReason: error`?

Reply:
the pointer should be from error to the provider log no? I honestly wouldn't add this pointer if we can avoid it it seems unnecessary

Reply:
Removed